src/datreant/core/collections.py

```diff
@@ -592,19 +595,26 @@ def __getitem__(self, index):
         # we can take lists of indices, names, or uuids; these return a
         # Bundle; repeats already not respected since Bundle functions as a
         # set
-        if ((isinstance(index, list) or hasattr(index, 'dtype')) and
-                all([isinstance(item, bool) for item in index])):
+        memberlist = self._list()
+
+        if hasattr(index, 'dtype'):  # detect numpy arrays without numpy import
+            if index.dtype == 'bool':
+                out = Bundle([memberlist[i]
+                              for i, val in enumerate(index) if val],
+                             limbs=self.limbs)
+            elif index.dtype == 'int':  # seems to pickup int64 just fine
```

Comment:
dtype is a special type with a comparision operator that lets this check pass for an array of any int type.